src/WebJobs.Script.WebHost/FileMonitoringService.cs

```diff
@@ -148,10 +148,15 @@ private void OnFileChanged(FileSystemEventArgs e)
                 // will be offline
                 // when app_offline.htm is deleted, we trigger
                 // a restart to bring the host back online
+                // we also need to reduce the debounce time here
+                // to avoid race condition causing the Host to
+                // not shutdown before a deployment starts
                 changeDescription = "File";
                 if (File.Exists(e.FullPath))
                 {
                     shutdown = true;
+                    _shutdown = Shutdown;
+                    _shutdown = _shutdown.Debounce(100);
```

Comment:
I just made up `100` ms (compared to `500` originally). Wondering if I should not even set any debounce here, or set it something like `10` ms.

Reply:
Do we want to debounce at all? Can we just call Shutdown() directly here?